File: Lintcode/82-Single-Number/SingleNumber.java

public class SingleNumber {
	/**
     *@param A : an integer array
     *return : a integer 
     */
	public int singleNumber(int[] nums) {
		int res = 0;
		if (nums == null || nums.length == 0) {
			return res;
		}
		for (int i : nums) {
			res = res ^ i;
		}
		return res;
	}
}

File: Leetcode/136-Single-Number/SingleNumber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    if (!nums || nums.length == 0) {
        return 0
    }
    var res =0
    for (i in nums) {
        res ^= nums[i]
    }
    return res
};


